airbyte-commons-worker/src/main/java/io/airbyte/workers/test_utils/AirbyteMessageUtils.java

```diff
@@ -102,29 +103,43 @@ public static AirbyteStreamState createStreamState(final String streamName) {
     return new AirbyteStreamState().withStreamDescriptor(new StreamDescriptor().withName(streamName));
   }
 
+  public static AirbyteMessage createEstimateMessage(final String name, final String namespace, final long byteEst, final long rowEst) {
+    final var est = new AirbyteEstimateTraceMessage()
+        .withType(AirbyteEstimateTraceMessage.Type.STREAM)
```

Comment:
@evantahler I realised there is also a `SYNC` type estimate message. What is this for?

I do not make a distinction in the processing logic between `SYNC` or `STREAM`. Should I?

Reply:
Yeah - Consider a CDC or Kafka source.  Until we actually read every Record, we don't know what stream it belongs to, but at the begning of the sync, we can get a whole-SYNC estimate that there are "100 rows to read".  In these cases, I think we can draw the overall progress bar but not the per-stream breakouts.   I would expect the per-stream APIs to be null, but the totals to be filled in.

Reply:
No source should be sending both estimates with STREAM and SYNC.  If you need to pick, choose STREAM information to win - and discard any SYNC estimates you have if a STREAM estimate appears.

Reply:
Descriptions added https://github.com/airbytehq/airbyte/pull/19843

Reply:
Thanks!